### PR TITLE
Add EXTENEnable property to AXP192

### PR DESCRIPTION
**Pull request by josesimoes**
## Description
- Add property to control state of EXTEN switch control.

## Motivation and Context

- Need to control this pin with M5Stack TOUGH.

## Types of changes

- [ ] Improvement (non-breaking change that improves a feature, code or algorithm)
- [ ] Bug fix (non-breaking change which fixes an issue with code or algorithm)
- [x] New feature (non-breaking change which adds functionality to code)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)
- [ ] Config and build (change in the configuration and build system, has no impact on code or features)
- [ ] Dependencies (update dependencies and changes associated, has no impact on code or features)
- [ ] Unit Tests (add new Unit Test(s) or improved existing one(s), has no impact on code or features)
- [ ] Documentation (changes or updates in the documentation, has no impact on code or features)

## Checklist:

- [x] My code follows the code style of this project (only if there are changes in source code).
- [x] My changes require an update to the documentation (there are changes that require the docs website to be updated).
- [ ] I have updated the documentation accordingly (the changes require an update on the docs in this repo).
- [x] I have read the [CONTRIBUTING](https://github.com/nanoframework/.github/blob/master/CONTRIBUTING.md) document.
- [ ] I have tested everything locally and all new and existing tests passed (only if there are changes in source code).
- [ ] I have added new tests to cover my changes.

### CODE REVIEW

**Comment by alberk8** [comment]
The ExtEn is on 0x10 Address.

image

**Comment by josesimoes** [comment]
@alberk8 thanks for the comment, but from the datasheet I have it looks like EXTEN should be accessed through register 0x12, not 0x10 (see note on table footer).
Also in M5Stack official library they are accessing register 0x12 too, not 0x10, so I'll go with 0x12 until this is proven wrong (as I don't have access to hardware right now to test it).
 
image

**Comment by alberk8** [comment]
> @alberk8 thanks for the comment, but from the datasheet I have it looks like EXTEN should be accessed through register 0x12, not 0x10 (see note on table footer). Also in M5Stack official library they are accessing register 0x12 too, not 0x10, so I'll go with 0x12 until this is proven wrong (as I don't have access to hardware right now to test it).
> 
> image

But When i looked at the M5Stack [code](https://github.com/m5stack/M5Core2/blob/0b905c583003eaccdf81c2f82b235f186903ca69/src/AXP192.cpp#L557) it does point to 0x10.   In fact I have tested it on nF and it seems to work.

**Comment by josesimoes** [comment]
What a mess! 🙄 

The TOUGH code [here](https://github.com/m5stack/M5Tough/blob/215fb2cee54ae351caeb658e4dd51811de969037/src/AXP192.cpp#L573) it's using 0x12...

And the AXP192 datasheet in TOUGH page [here](https://docs.m5stack.com/en/core/tough) (Chinese version only) is rev1.2.
From the comment on the table above (which seems to be added in rev1.2) I believe we can access that control signal from 0x12. 

And the code M5Stack support pointed to JorgeGOgreenYng it's also using 0x12. 

I'll go with 0x12 which is already here and let's wait for a confirmation.

File: devices/Axp192/Axp192.cs [hunk 1]
```diff
@@ -502,6 +503,33 @@ public void EnableDCDC1(bool state)
             I2cWrite(Register.SwitchControleDcDC1_3LDO2_3, buf);
         }
 
+        /// <summary>
+        /// Gets or Sets the state of EXTEN switch control.
+        /// </summary>
+        /// <value>
+        /// <see langword="true"/> if EXTEN switch is enabled, <see langword="false"/> otherwise.
+        /// </value>
+        public bool EXTENEnable
```
Comment (Ellerbach) [inline devices/Axp192/Axp192.cs:512]: Not the preper naming :-( Should have been `ExternEnable` or even better `EnableExtern`

**Comment by Ellerbach** [comment]
That's fine for this time but let's try to follow our proper naming pattern also with the acronyms

**Pull request merged**